### ISSUE TITLE
Can not get method_name from model files

### ISSUE THREAD

**Issue by tobegit3hub** [tensorflow/serving#369]
Hi, we have used `SavedModel` to export the TensorFlow model files and use `TensorFlow Serving` to load the files.

The server has started successfully.

```
I tensorflow_serving/model_servers/main.cc:144] Building single TensorFlow model file config:  model_name: mnist model_base_path: /home/tobe/code/serving/tensorflow_serving/example/mnist_model/ model_version_policy: 0
I tensorflow_serving/model_servers/server_core.cc:337] Adding/updating models.
I tensorflow_serving/model_servers/server_core.cc:383]  (Re-)adding model: mnist
I tensorflow_serving/core/basic_manager.cc:693] Successfully reserved resources to load servable {name: mnist version: 1}
I tensorflow_serving/core/loader_harness.cc:66] Approving load for servable version {name: mnist version: 1}
I tensorflow_serving/core/loader_harness.cc:74] Loading servable version {name: mnist version: 1}
I external/org_tensorflow/tensorflow/contrib/session_bundle/bundle_shim.cc:285] Attempting to load native SavedModelBundle in bundle-shim from: /home/tobe/code/serving/tensorflow_serving/example/mnist_model/1
I external/org_tensorflow/tensorflow/cc/saved_model/loader.cc:194] Loading SavedModel from: /home/tobe/code/serving/tensorflow_serving/example/mnist_model/1
W external/org_tensorflow/tensorflow/core/platform/cpu_feature_guard.cc:45] The TensorFlow library wasn't compiled to use SSE3 instructions, but these are available on your machine and could speed up CPU computations.
W external/org_tensorflow/tensorflow/core/platform/cpu_feature_guard.cc:45] The TensorFlow library wasn't compiled to use SSE4.1 instructions, but these are available on your machine and could speed up CPU computations.
W external/org_tensorflow/tensorflow/core/platform/cpu_feature_guard.cc:45] The TensorFlow library wasn't compiled to use SSE4.2 instructions, but these are available on your machine and could speed up CPU computations.
W external/org_tensorflow/tensorflow/core/platform/cpu_feature_guard.cc:45] The TensorFlow library wasn't compiled to use AVX instructions, but these are available on your machine and could speed up CPU computations.
W external/org_tensorflow/tensorflow/core/platform/cpu_feature_guard.cc:45] The TensorFlow library wasn't compiled to use AVX2 instructions, but these are available on your machine and could speed up CPU computations.
W external/org_tensorflow/tensorflow/core/platform/cpu_feature_guard.cc:45] The TensorFlow library wasn't compiled to use FMA instructions, but these are available on your machine and could speed up CPU computations.
I external/org_tensorflow/tensorflow/cc/saved_model/loader.cc:114] Restoring SavedModel bundle.
I external/org_tensorflow/tensorflow/cc/saved_model/loader.cc:148] Running LegacyInitOp on SavedModel bundle.
I external/org_tensorflow/tensorflow/cc/saved_model/loader.cc:238] Loading SavedModel: success. Took 42980 microseconds.
I tensorflow_serving/core/loader_harness.cc:86] Successfully loaded servable version {name: mnist version: 1}
I tensorflow_serving/model_servers/main.cc:256] Running ModelServer at 0.0.0.0:9000 ...
```

But we get the error when the client requested. It seems that the `method_name` is empty but we are sure that it was constructed as part of the protobuf object in Python.

```
Traceback (most recent call last):
  File "./predict_client.py", line 51, in <module>
    main()
  File "./predict_client.py", line 46, in main
    result = stub.Predict(request, request_timeout)
  File "/usr/local/lib/python2.7/dist-packages/grpc/beta/_client_adaptations.py", line 323, in __call__
    self._request_serializer, self._response_deserializer)
  File "/usr/local/lib/python2.7/dist-packages/grpc/beta/_client_adaptations.py", line 209, in _blocking_unary_unary
    raise _abortion_error(rpc_error_call)
grpc.framework.interfaces.face.face.AbortionError: AbortionError(code=StatusCode.INTERNAL, details="Expected prediction signature method_name to be one of {tensorflow/serving/predict, tensorflow/serving/classify, tensorflow/serving/regress}. Was: ")
```

**Comment by tobegit3hub** [comment]
By the way, the official `mnist_saved_model.py` will work in my environment but we don't know the difference between it and our models.

**Comment by terryKing1992** [comment]
perhaps pasting your exporting model code will be more helpful than the error;
in my point, maybe you dont set method_name for prediction signature object as follow:
```
prediction_signature = signature_def_utils.build_signature_def(
                    inputs={'input': tensor_info_x},
                    outputs={'output': tensor_info_y},
                    method_name=signature_constants.PREDICT_METHOD_NAME)
```

**Comment by tobegit3hub** [comment]
Thanks @terryKing1992 . The code of exporting the model  is like this.

```
model_signature = signature_def_utils.build_signature_def(
     inputs={
         'context_input':tensor_context_x,
         'query_input':tensor_query_x
     },
     outputs={
         'lstm_ouput':tensor_lstm_y,
         'cnn_output':tensor_cnn_y,
         'softmax_output':tensor_softmax_y
     },
     method_name=signature_constants.PREDICT_METHOD_NAME
)
```

We have tried reading the struct of signature protobuf and the string value is accessible.

```
print(model_signature.method_name)
```

**Comment by kirilg** [comment]
@tobegit3hub the signature export code you pasted looks right, I'm surprised the output doesn't include the method_name. When you inspect your output MetaGraphDef, what do the signatures look like? (For SavedModel models, the output also includes a string representation of the proto for easier inspection).

For example, this model has the following code snippet for a [prediction signature](https://github.com/tensorflow/tensorflow/blob/799e31f3840c21322e380e1ec6e5bacb95d016fa/tensorflow/examples/saved_model/saved_model_half_plus_two.py#L147) and the output [looks like this](https://github.com/tensorflow/tensorflow/blob/master/tensorflow/cc/saved_model/testdata/half_plus_two_pbtxt/00000123/saved_model.pbtxt#L2704).

**Comment by tobegit3hub** [comment]
Thanks @kirilg and we have checked the code again. The struct of `MetaGraphDef` has the `method_name` but it works when we try to export to an brand-new directory.

The cause of this problem may be that we had exported the `version 1` without the `method_name`. Then we added the `method_name` in the python code and try to export `version 2` and others which won't work.

It's fixed when we remove the problematic `version 1`. Thanks for the kind responses. @terryKing1992 @kirilg
